kotlin-dsl-layoutcontainer/src/main/java/com/hannesdorfmann/adapterdelegates4/dsl/LayoutContainerAdapterDelegateDsl.kt

```diff
@@ -0,0 +1,90 @@
+package com.hannesdorfmann.adapterdelegates4.dsl
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.annotation.LayoutRes
+import androidx.recyclerview.widget.RecyclerView
+import com.hannesdorfmann.adapterdelegates4.AbsItemAdapterDelegate
+import com.hannesdorfmann.adapterdelegates4.AdapterDelegate
+import com.hannesdorfmann.adapterdelegates4.AdapterItemProvider
+
+inline fun <reified I : T, T> adapterLayoutContainer(
+    @LayoutRes layout: Int,
+    noinline on: (item: T, items: AdapterItemProvider<T>, position: Int) -> Boolean = { item, _, _ -> item is I },
+    noinline layoutInflater: (parent: ViewGroup, layoutRes: Int) -> View = { parent, layoutRes ->
+        LayoutInflater.from(parent.context).inflate(
+            layoutRes,
+            parent,
+            false
+        )
+    },
+    noinline block: AdapterDelegateLayoutContainerViewHolder<I>.() -> Unit
+): AdapterDelegate<AdapterItemProvider<T>> {
+
+    return DslLayoutContainerAdapterDelegate(
+        layout = layout,
+        on = on,
+        initializerBlock = block,
+        layoutInflater = layoutInflater
+    )
+}
+
+@PublishedApi
+internal class DslLayoutContainerAdapterDelegate<I : T, T>(
+    @LayoutRes private val layout: Int,
+    private val on: (item: T, items: AdapterItemProvider<T>, position: Int) -> Boolean,
+    private val initializerBlock: AdapterDelegateLayoutContainerViewHolder<I>.() -> Unit,
+    private val layoutInflater: (parent: ViewGroup, layoutRes: Int) -> View
+) : AbsItemAdapterDelegate<I, T, AdapterDelegateLayoutContainerViewHolder<I>>() {
+
+    override fun isForViewType(item: T, items: AdapterItemProvider<T>, position: Int): Boolean = on(
+        item, items, position
```

Comment:
Here should be !=null check.
`    override fun isForViewType(item: T, items: AdapterItemProvider<T>, position: Int): Boolean {
        return item != null && on(item, items, position)
    }
`